### PR TITLE
feat: broadcast json serializable

### PR DESCRIPTION
**Pull request by luzhenqian**
The only changes at the api level are the parameters of the subscribe:

```js
channel.subscribe('eventName', (payload, state) => {
  // ...
})
```

Other changes are that json strings are used internally for data transmission, and there is no change at the api level.

### CODE REVIEW

File: packages/presence/src/channel.ts [hunk 1]
```diff
@@ -56,19 +58,23 @@ export class Channel implements IChannel {
...
-    );
+  broadcast<T extends JsonSerializable>(eventName: string, payload: T): void {
+    try {
+      const _payload = JsonSerializer.serialize(payload)
+
+      this.#broadcast(
+        eventName,
+        this.#getDataPacket({
+          payload: _payload
+        })
```
Comment (fanweixiao) [inline packages/presence/src/channel.ts:69]:
```suggestion
        data: _payload
```

as the discussion: https://github.com/orgs/xileteam/discussions/672

**Review by fanweixiao — CHANGES_REQUESTED** [review]
confused with what we discussed:  https://github.com/orgs/xileteam/projects/16/views/1?pane=issue&itemId=37850808

**Comment by fanweixiao** [comment]
I think should be:

```js
channel.subscribe('eventName', (payload, peer) => { // state -> peer
  // ...
})
```

**Review by fanweixiao — APPROVED** [review]

File: packages/presence/src/types.d.ts [hunk 2]
```diff
@@ -84,10 +83,20 @@ export type ChannelEventSubscribeCallbackFn<T> = (
...
+type JsonPrimitive = string | number | boolean | null;
+
+interface JsonArray extends Array<JsonSerializable> { }
+
+interface JsonObject {
+  [key: string]: JsonSerializable;
+}
+
+export type JsonSerializable = JsonPrimitive | JsonObject | JsonArray;
+
```
[the remark this answers is not in the document]
Comment (luzhenqian) [reply packages/presence/src/types.d.ts:90]: `undefined` is not standard JSON:

image

Advantages of my writing style:

More modular: By using separate types and interfaces (such as JsonPrimitive, JsonArray, and JsonObject), this way of writing provides better readability and organization.
Extensibility: If you need to modify or add annotations to a subtype in the future, the modular design can make this process easier.

**Pull request merged**